py/dml/template.py

```diff
@@ -136,9 +136,9 @@ def site(self): return self.spec.site
     def traits(self):
         '''Return a set of all traits implemented by this template'''
         if self.trait:
-            return {self.trait} | self.trait.ancestors
+            return Set((self.trait, *self.trait.ancestors))
         else:
-            return set().union(
+            return Set().union(
                 *[sup.traits() for (_, sup) in self.spec.templates])
```

Comment:
assuming you've checked that self.spec.templates is deterministic here

Reply:
Yeah, `ObjectSpec.templates` is always a list.